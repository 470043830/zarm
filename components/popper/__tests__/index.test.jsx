import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Popper from '../index';

describe('Popper', () => {
  it('renders correctly', () => {
    const onVisibleChange = jest.fn();

    const wrapper = mount(<Popper title="标题" onVisibleChange={onVisibleChange}><p>点我</p></Popper>);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
