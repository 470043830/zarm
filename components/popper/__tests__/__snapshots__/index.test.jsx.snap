// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popper renders correctly 1`] = `
<Popper
  content=""
  direction="top"
  hasArrow={false}
  mouseEnterDelay={100}
  mouseLeaveDelay={100}
  onVisibleChange={[MockFunction]}
  prefixCls="za-popper"
  title="标题"
  trigger="hover"
  visible={false}
>
  <div
    className="za-popper"
  >
    <div
      className="za-popper__inner"
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
    >
      <p>
        点我
      </p>
    </div>
  </div>
</Popper>
`;
